src/function/maybe_changed_after.rs

```diff
@@ -408,57 +372,49 @@ where
                 // Conditionally specified queries
                 // where the value is specified
                 // in rev 1 but not in rev 2.
-                VerifyResult::changed()
+                VerifyResult::Changed
             }
             // Return `Unchanged` similar to the initial value that we insert
             // when we hit the cycle. Any dependencies accessed when creating the fixpoint initial
             // are tracked by the outer query. Nothing should have changed assuming that the
             // fixpoint initial function is deterministic.
-            QueryOrigin::FixpointInitial => VerifyResult::Unchanged(
-                InputAccumulatedValues::Empty,
-                CycleHeads::initial(database_key_index),
-            ),
+            QueryOrigin::FixpointInitial => VerifyResult::unchanged(),
```

Comment:
I am not 100% sure but I believe we do not need to push `database_key_index` into the cycle heads here, we only need to extend it with the dependency cycle heads, but `FixpointInitial` has no relevant dependencies (unless the initial kicks off a cycle itself which iirc is not supported scenario).

Reply:
We need to push the cycle head here. We've seen hangs without it. My reasoning behind this is that it otherwise behaves differently from when we push an initial value when we encounter the cycle the first time. 

In the end, this still signifies a cycle and we can only backdate the cycle head (and not the current query that only participates in the cycle)